server/lib/src/plugins/dyngroup.rs

```diff
@@ -208,22 +230,53 @@ impl DynGroup {
                         .copied()
                         .for_each(|u| d_group.add_ava(Attribute::DynMember, Value::Refer(u)));
 
-                    affected_uuids.extend(matches.into_iter());
-                    affected_uuids.push(*dg_uuid);
+                    let pre_dynmember = pre.get_ava_refer(Attribute::DynMember);
+                    let post_dynmember = d_group.get_ava_refer(Attribute::DynMember);
+
+                    match (pre_dynmember, post_dynmember) {
+                        (Some(pre_m), Some(post_m)) => {
+                            // Show only the *changed* uuids.
+                            affected_uuids.extend(pre_m.symmetric_difference(post_m));
+                        }
+                        (Some(members), None) | (None, Some(members)) => {
+                            // Doesn't matter what order, just that they are affected
+                            affected_uuids.extend(members);
+                        }
+                        (None, None) => {}
+                    };
+
+                    // The *dyn group* isn't changing, it's that a member OF the dyn group
+                    // is being added. This means the dyngroup isn't part of the set that
+                    // needs update to MO, only the affected members do!
+                    /*
+                    if pre.get_ava_set(Attribute::DynMember)
+                        != d_group.get_ava_set(Attribute::DynMember)
+                    {
+                        affected_uuids.insert(*dg_uuid);
+                    }
+                    */
```

Comment:
Redundant now? 
```suggestion
```

Reply:
I left it there as a "just in case", bunt I guess it can just be removed. I still think we need the comment though.